pipenv/routines/outdated.py

```diff
@@ -1,19 +1,16 @@
 import sys
+from collections import namedtuple
+from collections.abc import Mapping
 
+from pipenv.patched.pip._vendor.packaging.utils import canonicalize_name
 from pipenv.routines.lock import do_lock
 from pipenv.utils.dependencies import pep423_name
 from pipenv.vendor import click
+from pipenv.vendor.requirementslib.models.requirements import Requirement
+from pipenv.vendor.requirementslib.models.utils import get_version
```

Comment:
Neat: removing in function imports to top level because of linter isn't something I favour.
Top level imports slow down the start time of programs. For command line tools which might have short execution time, this might be significant.
This might not have a huge impact here, though.